test/steps/v2_steps.py

```diff
@@ -2445,10 +2515,25 @@ def append_app_args_to_method_args(context, method_args):
     context.method_args += app_args
 
 
-@step(
-    'I add a method call with the {account_type} account, the current application, suggested params, on complete "{operation}", current transaction signer, current method arguments.'
-)
-def add_abi_method_call(context, account_type, operation):
+@given('I add the nonce "{nonce}"')
+def add_nonce(context, nonce):
+    context.nonce = nonce
+
+
+def abi_method_adder(
```

Comment:
Get's a little hard to read hear. I refactored the method call steps to allow for the nonce and for passing app-ids from the context as method args